SofaKernel/modules/SofaSimpleFem/src/SofaSimpleFem/TriangleFEMUtils.inl

```diff
@@ -0,0 +1,284 @@
+/******************************************************************************
+*                 SOFA, Simulation Open-Framework Architecture                *
+*                    (c) 2006 INRIA, USTL, UJF, CNRS, MGH                     *
+*                                                                             *
+* This program is free software; you can redistribute it and/or modify it     *
+* under the terms of the GNU Lesser General Public License as published by    *
+* the Free Software Foundation; either version 2.1 of the License, or (at     *
+* your option) any later version.                                             *
+*                                                                             *
+* This program is distributed in the hope that it will be useful, but WITHOUT *
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or       *
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License *
+* for more details.                                                           *
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#pragma once
+#include <SofaSimpleFem/TriangleFEMUtils.h>
+
+namespace sofa::component::forcefield
+{
+
+
+////////////// small displacements method
+
+// ---------------------------------------------------------------------------------------------------------------
+// ---	Compute displacement vector D as the difference between current current position 'p' and initial position
+// ---------------------------------------------------------------------------------------------------------------
+template<class DataTypes>
+constexpr void TriangleFEMUtils<DataTypes>::computeDisplacementSmall(Displacement& D, const type::fixed_array<Coord, 3>& rotatedInitCoord, const Coord& pAB, const Coord& pAC)
+{
+    // First vector: deforme_a = pA - pA = Coord(0,0,0);
+    // deforme_b = pB - pA == pAB
+    // deforme_c = pB - pA == pAC
+
+    D[0] = 0;
+    D[1] = 0;
+    D[2] = rotatedInitCoord[1][0] - pAB[0];
+    D[3] = rotatedInitCoord[1][1] - pAB[1];
+    D[4] = rotatedInitCoord[2][0] - pAC[0];
+    D[5] = rotatedInitCoord[2][1] - pAC[1];
+}
+
+
+////////////// large displacements method
+
+template<class DataTypes>
+constexpr void TriangleFEMUtils<DataTypes>::computeRotationLarge(Transformation& r, const Coord& pA, const Coord& pB, const Coord& pC)
```

Comment:
how can this be a constexpr ?

Reply:
constexpr is inline by nature 
([dcl.constexpr], §7.1.5/2 in the C++11 standard): "constexpr functions and constexpr constructors are implicitly inline (7.1.2)."